File: iOS/LittleStoneBookkeeping/MainViewController.m
#import "MainViewController.h"
#import "StoneBookkeepingViewController.h"
#import "AndroidMirrorViewController.h"
#import "FeedbackViewController.h"

typedef NS_ENUM(NSInteger, InterfaceType) {
    InterfaceTypeStoneBookkeeping,
    InterfaceTypeAndroidMirror
};

@interface MainViewController ()

@property (nonatomic, assign) InterfaceType currentInterface;
@property (nonatomic, strong) UILabel *titleLabel;
@property (nonatomic, strong) UIButton *switchButton;
@property (nonatomic, strong) UIView *containerView;

@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.currentInterface = InterfaceTypeStoneBookkeeping;
    
    [self setupUI];
    [self setupStoneBookkeepingInterface];
    
    // 监听界面切换通知
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(handleInterfaceSwitch)
                                                 name:@"SwitchToBInterface"
                                               object:nil];
}

- (void)dealloc {
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}

- (void)handleInterfaceSwitch {
    [self performInterfaceSwitch];
}

- (void)setupUI {
    self.view.backgroundColor = [UIColor systemBackgroundColor];
    
    // 标题
    self.titleLabel = [[UILabel alloc] init];
    self.titleLabel.text = @"小石头记账";
    self.titleLabel.font = [UIFont boldSystemFontOfSize:24];
    self.titleLabel.textAlignment = NSTextAlignmentCenter;
    
    // 切换按钮
    self.switchButton = [UIButton buttonWithType:UIButtonTypeSystem];
    [self.switchButton setTitle:@"切换界面" forState:UIControlStateNormal];
    [self.switchButton addTarget:self action:@selector(switchInterface) forControlEvents:UIControlEventTouchUpInside];
    
    // 容器视图
    self.containerView = [[UIView alloc] init];
    
    // 布局
    UIStackView *stackView = [[UIStackView alloc] initWithArrangedSubviews:@[self.titleLabel, self.switchButton, self.containerView]];
    stackView.axis = UILayoutConstraintAxisVertical;
    stackView.spacing = 20;
    stackView.translatesAutoresizingMaskIntoConstraints = NO;
    
    [self.view addSubview:stackView];
    
    [NSLayoutConstraint activateConstraints:@[
        [stackView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor constant:20],
        [stackView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor constant:20],
        [stackView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor constant:-20],
        [stackView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor constant:-20]
    ]];
    
    // 为容器视图添加高度约束，确保它有足够的空间显示内容
    [NSLayoutConstraint activateConstraints:@[
        [self.containerView.heightAnchor constraintGreaterThanOrEqualToConstant:400]
    ]];
}

- (void)switchInterface {
    // 显示意见反馈界面进行切换
    FeedbackViewController *feedbackVC = [[FeedbackViewController alloc] init];
    __weak typeof(self) weakSelf = self;
    feedbackVC.interfaceSwitchHandler = ^{
        [weakSelf performInterfaceSwitch];
    };
    [self presentViewController:feedbackVC animated:YES completion:nil];
}

- (void)performInterfaceSwitch {
    // 清除当前界面
    for (UIView *subview in self.containerView.subviews) {
        [subview removeFromSuperview];
    }
    
    switch (self.currentInterface) {
        case InterfaceTypeStoneBookkeeping:
            self.currentInterface = InterfaceTypeAndroidMirror;
            self.titleLabel.text = @"答题大师";
            [self setupAndroidMirrorInterface];
            break;
        case InterfaceTypeAndroidMirror:
            self.currentInterface = InterfaceTypeStoneBookkeeping;
            self.titleLabel.text = @"小石头记账";
            [self setupStoneBookkeepingInterface];
            break;
    }
}

- (void)setupStoneBookkeepingInterface {
    // 创建记账界面
    StoneBookkeepingViewController *bookkeepingVC = [[StoneBookkeepingViewController alloc] init];
    [self addChildViewController:bookkeepingVC];
    [self.containerView addSubview:bookkeepingVC.view];
    bookkeepingVC.view.translatesAutoresizingMaskIntoConstraints = NO;
    [bookkeepingVC didMoveToParentViewController:self];
    
    [NSLayoutConstraint activateConstraints:@[
        [bookkeepingVC.view.topAnchor constraintEqualToAnchor:self.containerView.topAnchor],
        [bookkeepingVC.view.leadingAnchor constraintEqualToAnchor:self.containerView.leadingAnchor],
        [bookkeepingVC.view.trailingAnchor constraintEqualToAnchor:self.containerView.trailingAnchor],
        [bookkeepingVC.view.bottomAnchor constraintEqualToAnchor:self.containerView.bottomAnchor]
    ]];
}

- (void)setupAndroidMirrorInterface {
    // 检查是否是第二次登录
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    BOOL isSecondLogin = [defaults boolForKey:@"has_quiz_loaded"];
    
    if (isSecondLogin) {
        // 第二次登录，直接显示答题界面
        [self showAndroidMirrorInterface];
    } else {
        // 第一次登录，显示加载中界面
        [self showLoadingInterface];
    }
}

- (void)showLoadingInterface {
    // 清除当前界面
    for (UIView *subview in self.containerView.subviews) {
        [subview removeFromSuperview];
    }
    
    // 创建加载中界面
    UIView *loadingView = [[UIView alloc] init];
    loadingView.backgroundColor = [UIColor systemBackgroundColor];
    
    UILabel *loadingLabel = [[UILabel alloc] init];
    loadingLabel.text = @"加载中...";
    loadingLabel.font = [UIFont boldSystemFontOfSize:20];
    loadingLabel.textAlignment = NSTextAlignmentCenter;
    
    UIProgressView *progressView = [[UIProgressView alloc] initWithProgressViewStyle:UIProgressViewStyleDefault];
    progressView.progress = 0.0;
    
    UIStackView *loadingStack = [[UIStackView alloc] initWithArrangedSubviews:@[loadingLabel, progressView]];
    loadingStack.axis = UILayoutConstraintAxisVertical;
    loadingStack.spacing = 20;
    loadingStack.translatesAutoresizingMaskIntoConstraints = NO;
    
    [loadingView addSubview:loadingStack];
    [self.containerView addSubview:loadingView];
    
    [NSLayoutConstraint activateConstraints:@[
        [loadingView.topAnchor constraintEqualToAnchor:self.containerView.topAnchor],
        [loadingView.leadingAnchor constraintEqualToAnchor:self.containerView.leadingAnchor],
        [loadingView.trailingAnchor constraintEqualToAnchor:self.containerView.trailingAnchor],
        [loadingView.bottomAnchor constraintEqualToAnchor:self.containerView.bottomAnchor],
        
        [loadingStack.centerXAnchor constraintEqualToAnchor:loadingView.centerXAnchor],
        [loadingStack.centerYAnchor constraintEqualToAnchor:loadingView.centerYAnchor],
        [loadingStack.leadingAnchor constraintEqualToAnchor:loadingView.leadingAnchor constant:40],
        [loadingStack.trailingAnchor constraintEqualToAnchor:loadingView.trailingAnchor constant:-40]
    ]];
    
    // 模拟15秒加载过程
    __block float progress = 0.0;
    NSTimer *timer = [NSTimer scheduledTimerWithTimeInterval:0.1 repeats:YES block:^(NSTimer * _Nonnull timer) {
        progress += 0.00667; // 15秒完成
        progressView.progress = progress;
        
        if (progress >= 1.0) {
            [timer invalidate];
            // 加载完成，标记为已加载
            NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
            [defaults setBool:YES forKey:@"has_quiz_loaded"];
            // 显示答题界面
            [self showAndroidMirrorInterface];
        }
    }];
}

- (void)showAndroidMirrorInterface {
    // 清除当前界面
    for (UIView *subview in self.containerView.subviews) {
        [subview removeFromSuperview];
    }
    
    // 创建Android镜像界面
    AndroidMirrorViewController *androidVC = [[AndroidMirrorViewController alloc] init];
    [self addChildViewController:androidVC];
    [self.containerView addSubview:androidVC.view];
    androidVC.view.translatesAutoresizingMaskIntoConstraints = NO;
    [androidVC didMoveToParentViewController:self];
    
    [NSLayoutConstraint activateConstraints:@[
        [androidVC.view.topAnchor constraintEqualToAnchor:self.containerView.topAnchor],
        [androidVC.view.leadingAnchor constraintEqualToAnchor:self.containerView.leadingAnchor],
        [androidVC.view.trailingAnchor constraintEqualToAnchor:self.containerView.trailingAnchor],
        [androidVC.view.bottomAnchor constraintEqualToAnchor:self.containerView.bottomAnchor]
    ]];
}

@end

File: iOS/LittleStoneBookkeeping/MainViewController.swift
import UIKit

class MainViewController: UIViewController {
    
    // 界面状态管理
    private var currentInterface: InterfaceType = .stoneBookkeeping
    
    // 界面类型枚举
    private enum InterfaceType {
        case stoneBookkeeping  // A面：小石头记账
        case androidMirror     // B面：Android镜像界面
    }
    
    // 记账相关属性
    private var transactions: [Transaction] = []
    private var categories: [Category] = [
        Category(id: 1, name: "餐饮", icon: "🍽️"),
        Category(id: 2, name: "交通", icon: "🚗"),
        Category(id: 3, name: "购物", icon: "🛍️"),
        Category(id: 4, name: "娱乐", icon: "🎮"),
        Category(id: 5, name: "医疗", icon: "🏥"),
        Category(id: 6, name: "教育", icon: "📚"),
        Category(id: 7, name: "收入", icon: "💰")
    ]
    
    // UI组件
    private let titleLabel = UILabel()
    private let switchButton = UIButton(type: .system)
    private let containerView = UIView()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
        setupStoneBookkeepingInterface()
        
        // 监听界面切换通知
        NotificationCenter.default.addObserver(
            self,
            selector: #selector(handleInterfaceSwitch),
            name: NSNotification.Name("SwitchToBInterface"),
            object: nil
        )
    }
    
    @objc private func handleInterfaceSwitch() {
        performInterfaceSwitch()
    }
    
    private func setupUI() {
        view.backgroundColor = .systemBackground
        
        // 标题
        titleLabel.text = "小石头记账"
        titleLabel.font = UIFont.boldSystemFont(ofSize: 24)
        titleLabel.textAlignment = .center
        
        // 切换按钮
        switchButton.setTitle("切换界面", for: .normal)
        switchButton.addTarget(self, action: #selector(switchInterface), for: .touchUpInside)
        
        // 布局
        let stackView = UIStackView(arrangedSubviews: [titleLabel, switchButton, containerView])
        stackView.axis = .vertical
        stackView.spacing = 20
        stackView.translatesAutoresizingMaskIntoConstraints = false
        
        view.addSubview(stackView)
        
        NSLayoutConstraint.activate([
            stackView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 20),
            stackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
            stackView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
            stackView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -20)
        ])
    }
    
    @objc private func switchInterface() {
        // 显示意见反馈界面进行切换
        let feedbackVC = FeedbackViewController()
        feedbackVC.interfaceSwitchHandler = { [weak self] in
            self?.performInterfaceSwitch()
        }
        present(feedbackVC, animated: true)
    }
    
    private func performInterfaceSwitch() {
        // 清除当前界面
        containerView.subviews.forEach { $0.removeFromSuperview() }
        
        switch currentInterface {
        case .stoneBookkeeping:
            currentInterface = .androidMirror
            titleLabel.text = "答题大师"
            setupAndroidMirrorInterface()
        case .androidMirror:
            currentInterface = .stoneBookkeeping
            titleLabel.text = "小石头记账"
            setupStoneBookkeepingInterface()
        }
    }
    
    private func setupStoneBookkeepingInterface() {
        // 创建记账界面
        let bookkeepingVC = StoneBookkeepingViewController()
        addChild(bookkeepingVC)
        containerView.addSubview(bookkeepingVC.view)
        bookkeepingVC.view.translatesAutoresizingMaskIntoConstraints = false
        bookkeepingVC.didMove(toParent: self)
        
        NSLayoutConstraint.activate([
            bookkeepingVC.view.topAnchor.constraint(equalTo: containerView.topAnchor),
            bookkeepingVC.view.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
            bookkeepingVC.view.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
            bookkeepingVC.view.bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
        ])
    }
    
    private func setupAndroidMirrorInterface() {
        // 检查是否是第二次登录
        let defaults = UserDefaults.standard
        let isSecondLogin = defaults.bool(forKey: "has_quiz_loaded")
        
        if isSecondLogin {
            // 第二次登录，直接显示答题界面
            showAndroidMirrorInterface()
        } else {
            // 第一次登录，显示加载中界面
            showLoadingInterface()
        }
    }
    
    private func showLoadingInterface() {
        // 清除当前界面
        containerView.subviews.forEach { $0.removeFromSuperview() }
        
        // 创建加载中界面
        let loadingView = UIView()
        loadingView.backgroundColor = .systemBackground
        
        let loadingLabel = UILabel()
        loadingLabel.text = "加载中..."
        loadingLabel.font = UIFont.boldSystemFont(ofSize: 20)
        loadingLabel.textAlignment = .center
        
        let progressView = UIProgressView(progressViewStyle: .default)
        progressView.progress = 0.0
        
        let loadingStack = UIStackView(arrangedSubviews: [loadingLabel, progressView])
        loadingStack.axis = .vertical
        loadingStack.spacing = 20
        loadingStack.translatesAutoresizingMaskIntoConstraints = false
        
        loadingView.addSubview(loadingStack)
        containerView.addSubview(loadingView)
        
        NSLayoutConstraint.activate([
            loadingView.topAnchor.constraint(equalTo: containerView.topAnchor),
            loadingView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
            loadingView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
            loadingView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
            
            loadingStack.centerXAnchor.constraint(equalTo: loadingView.centerXAnchor),
            loadingStack.centerYAnchor.constraint(equalTo: loadingView.centerYAnchor),
            loadingStack.leadingAnchor.constraint(equalTo: loadingView.leadingAnchor, constant: 40),
            loadingStack.trailingAnchor.constraint(equalTo: loadingView.trailingAnchor, constant: -40)
        ])
        
        // 模拟15秒加载过程
        var progress: Float = 0.0
        let timer = Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { timer in
            progress += 0.00667 // 15秒完成
            progressView.progress = progress
            
            if progress >= 1.0 {
                timer.invalidate()
                // 加载完成，标记为已加载
                let defaults = UserDefaults.standard
                defaults.set(true, forKey: "has_quiz_loaded")
                // 显示答题界面
                self.showAndroidMirrorInterface()
            }
        }
    }
    
    private func showAndroidMirrorInterface() {
        // 清除当前界面
        containerView.subviews.forEach { $0.removeFromSuperview() }
        
        // 创建Android镜像界面
        let androidVC = AndroidMirrorViewController()
        addChild(androidVC)
        containerView.addSubview(androidVC.view)
        androidVC.view.translatesAutoresizingMaskIntoConstraints = false
        androidVC.didMove(toParent: self)
        
        NSLayoutConstraint.activate([
            androidVC.view.topAnchor.constraint(equalTo: containerView.topAnchor),
            androidVC.view.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
            androidVC.view.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
            androidVC.view.bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
        ])
    }
}

// 数据模型
struct Transaction {
    let id: UUID
    let amount: Double
    let category: Category
    let date: Date
    let note: String
    let type: TransactionType
}

enum TransactionType {
    case income
    case expense
}

struct Category {
    let id: Int
    let name: String
    let icon: String
}